include/ensmallen_bits/cmaes/transformation_policies/boundary_box_constraint.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file boundary_box_constraint.hpp
+ * @author Suvarsha Chennareddy
+ *
+ * Boundary Box Transformation.
+ *
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+namespace ens {
+  /**
+  * More often than not, coordinates must be bounded by some constraints.
+  * In a particular case, the domain of a specific function is restricted 
+  * by boundaries.
+  * The implemented transformation transforms given coordinates into a region 
+  * bounded by the given lower and upper bounds (a box). First, the 
+  * coordinates are shifted into a feasible preimage bounded by lower_bound - al 
+  * and upper_bound + au where al and au and calculated internally. 
+  * These shifted coordinates are then transformed into coordinates bounded by 
+  * lower_bound and upper_bound. It is an identity transformation in between 
+  * the lower and upper bounds.
+  * The inverse transformation transforms the coordinates back to the preimage.
+  * 
+  * For more information, check the original implementation in C by N. Hansen:
+  * https://github.com/CMA-ES/c-cmaes/blob/master/src/boundary_transformation.c
+  *
+  * @tparam MatType The matrix type of the coordinates and bounds.
+  */
+template<typename MatType = arma::mat>
+class BoundaryBoxConstraint
+{
+public:
+  /**
+  * Construct the boundary box constraint poilcy.
+  *
+  * @param lowerBound The lower bound of the coordinates.
+  * @param upperBound The upper bound of the coordinates.
+  */
+  BoundaryBoxConstraint(const MatType& lowerBound,
+    const MatType& upperBound):
```

Comment:
So, here you are allowing a different lower and upper bound for each element of the input?  My opinion is we should also provide a `double` constructor that reflects the previous functionality (so every dimension's limits are `[lowerBound, upperBound]`).

Reply:
Alright! This would work well with preserving a constructor that takes `lowerBound` and `upperBound` as parameters of type `double.`